content/blog/vdiffr-1-0-0/index.Rmd

```diff
@@ -0,0 +1,104 @@
+
+photo:
+  url: https://unsplash.com/photos/e8rfcKAx1Rk
+  author: Jakob Owens
+
+categories: [package] 
+tags: []
+---
+
+We're delighted to announce the release of [vdiffr](https://vdiffr.r-lib.org/) 1.0.0. vdiffr is a testthat extension that makes it easy to automatically check code that generates R graphics. In particular, vdiffr is used by the ggplot2 team to ensure that changes and contributions do not affect the expected output of plots.
+
+You can install vdiffr from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("vdiffr")
+```
+
+This blog post briefly introduces the vdiffr workflow and describes the changes in this 1.0 version. In the last section you will learn how to migrate your existing vdiffr snapshots to the new version. You can see a full list of changes in the [release notes](https://vdiffr.r-lib.org/news/index.html#vdiffr-1-0-0-2021-06-08).
+
+Attach these three packages to follow the examples:
+
+```{r setup}
+library(testthat)
+library(vdiffr)
+library(ggplot2)
+```
+
+
+## Regression testing for R graphics
+
+vdiffr is a [testthat](https://testthat.r-lib.org/) extension for monitoring the appearance of R plots and graphics over time. Its goals are to make it easy to test graphics, easy to review changes, and to be reproducible across platforms.
+
+The only vdiffr function you will need to use is `expect_doppelganger()`. It takes a plot title and a plot object.
+
+```{r, eval = FALSE}
+test_that("ggplot2 histogram works", {
+  p <- ggplot(mtcars) + geom_histogram(aes(disp))
+  expect_doppelganger("default histogram", p)
+})
+```
+
+With base graphics you will need to use a slightly different syntax because base plots are created by side effects rather than a plot object as in ggplot2. In this case you can supply a function that generates the plot:
+
+```{r, eval = FALSE}
+test_that("base histogram works", {
+  p <- function() hist(mtcars$disp)
+  expect_doppelganger("base histogram", p)
+})
+```
+
+Then run `devtools::test()` as usual. The first time it is run, `expect_doppelganger()` generates a reproducible SVG file that represents the expected appearance of your plot and saves it inside your test folder. After that, the generated SVG is compared to the saved version and any mismatch is reported as a failure in the testthat output.
+
+By default though, mismatches do not cause failures on CRAN machines. That's because testing the appearance of a plot is inherently fragile. Small upstream changes (e.g. in the R graphics device or in the ggplot2 package) might cause subtle differences in the appearance of a plot that are not real failures. If every such change caused a CRAN failure this would be distracting for both you and the CRAN maintainers. This is why vdiffr expectations only report failures when they are ran locally or in your CI platform. This allows you to monitor the appearance of your plots over time without causing distracting and time-consuming failures on CRAN.
```

Comment:
```suggestion
Note that, by default, mismatches do not cause failures on CRAN machines. That's because testing the appearance of a plot is inherently fragile. Small upstream changes (e.g. in the R graphics device or in the ggplot2 package) might cause subtle differences in the appearance of a plot that are not real failures. If every such change caused a CRAN failure this would be distracting for both you and the CRAN maintainers. This is why vdiffr expectations only report failures when they are ran locally or in your CI platform. This allows you to monitor the appearance of your plots over time without causing distracting and time-consuming failures on CRAN.
```